src/addressResolvers/farcaster.ts

```diff
@@ -0,0 +1,151 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { graphQlCall, Address, Handle, FetchError, isSilencedError, isEvmAddress } from './utils';
+import { getAddress } from 'viem';
+
+export const NAME = 'Farcaster';
+const FNAMES_API_URL = 'https://fnames.farcaster.xyz/transfers?name=';
+const NEYNAR_API_URL = 'https://api.neynar.com/v2/farcaster/user/';
+const API_KEY = process.env.NEYNAR_API_KEY ?? '';
+
+interface UserDetails {
+  username: Handle;
+  verified_addresses: {
+    eth_addresses: Address[];
+    sol_addresses: string[];
+  };
+  pfp_url: string;
+}
+
+interface ApiResponse {
+  [address: string]: UserDetails[];
+}
+
+interface UserResult {
+  username?: Handle;
+  eth_addresses?: Address[];
+  sol_addresses?: string[];
+  pfp_url?: string;
+}
+
+export async function fetchData<T>(url: string): Promise<T> {
+  const headers = {
+    Accept: 'application/json',
+    api_key: API_KEY
+  };
+  const response = await fetch(url, { headers });
+  if (!response.ok) {
+    const error = new FetchError(`Failed to fetch data from the API. Status: ${response.status}`);
+    if (!isSilencedError(error)) {
+      capture(error, { tags: { issue: 'api_fetch_failure' } });
+    }
+    throw error;
+  }
+  return response.json() as Promise<T>;
+}
+
+
+export async function lookupAddresses(addresses: Address[]): Promise<{ [key: Handle]: Address }> {
```

Comment:
Please move this function to file bottom, to have all exported function together